.circleci/config.yml

```diff
@@ -0,0 +1,39 @@
+# We are supposed to use architect-orb to build&push container images.
+# However, architect-orb has strict rules and tests. We don't have full control over this repository.
+# That is why we introduced this custom configuration in this repository.
+# Keep this file as it is while merging changes from upstream repository.
+
+version: 2.1
+
+jobs:
+  build:
+    machine:
+      image: ubuntu-2004:2024.04.4
+    steps:
+      - checkout
+      - run:
+          name: Print env vars
+          command: |
+            echo "CIRCLE_TAG=${CIRCLE_TAG}"
+      - run:
+          name : Build container
+          command: |
+            make build
+            make ccm
+      - run:
+          name: Push container
+          command: |
+            echo "$QUAY_PASSWORD" | docker login quay.io --username $QUAY_USERNAME --password-stdin
```

Comment:
@anvddriesch Could you please replace this with gsoci.azurecr.io?

Reply:
hint: https://github.com/giantswarm/cluster-api-provider-azure/commit/f6d13b178f55897f2ebd3a96e6e7479158727ca7#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R30

Reply:
Damn I thought this was taken of by crsync but it does seem like the tag isn't present in the azure cr. I made the change